assignments/prototype-refactor.js

```diff
@@ -1,9 +1,231 @@
-/* 
+/*
 
 Prototype Refactor
 
 1. Copy and paste your code or the solution from yesterday
 
-2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
+2. Your goal is to refactor all of this code to use ES6 Classes.
+The console.log() statements should still return what is expected of them.
 
 */
+/*
+  === GameObject ===
+  * createdAt
+  * name
+  * dimensions (These represent the character's size in the video game)
+  * destroy() // prototype method that returns: `${this.name} was removed from the game.`
+*/
+
+class GameObject {
+  constructor(param) {
+    this.createdAt = param.createdAt,
+    this.name = param.name,
+    this.dimensions = param.dimensions
+  }
+  destroy() {
+    return `${this.name} was removed from the game`;
+  }
+}
+
+/*
+  === CharacterStats ===
+  * healthPoints
+  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+  * should inherit destroy() from GameObject's prototype
+*/
+class CharacterStats extends GameObject {
+  constructor(stats) {
+    super(stats);
+    this.healthPoints = stats.healthPoints
+  }
+  takeDamage() {
+    return `${this.name} took damage`;
+ }
+}
+
+/*
+  === Humanoid (Having an appearance or character resembling that of a human.) ===
+  * team
+  * weapons
+  * language
+  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+  * should inherit destroy() from GameObject through CharacterStats
+  * should inherit takeDamage() from CharacterStats
+*/
+
+class Humanoid extends CharacterStats {
+  constructor(attr) {
+    super(attr);
+    this.team = attr.team,
+    this.weapons = attr.weapons,
+    this.language = attr.language
+  }
+  greet() {
+    return `${this.name} offers a greeting in ${this.language}.`
+  }
+}
+
+
+class Hero extends Humanoid {
+  constructor(attr) {
+    super(attr);
+    this.attack = attr.attack
+  }
+  death() {
+    if (this.healthPoints <= 0) {
+      return "you are dead";
+    } else {
+      return 'you are alive!'
```

Comment:
Based on what this method does (it seems to check if the `Hero` is dead or alive), you might want to rename it to reflect its behavior (some suggestions for renaming it: `.isAlive`, `.isDead`, `.checkLiving`, `.checkStatus`, etc.).